internal/bminventory/inventory.go

```diff
@@ -2543,6 +2543,17 @@ func (b *bareMetalInventory) updatePlatformParams(params installer.V2UpdateClust
 	if params.ClusterUpdateParams.Platform != nil && common.PlatformTypeValue(params.ClusterUpdateParams.Platform.Type) != "" {
 		updates["platform_type"] = params.ClusterUpdateParams.Platform.Type
 		updates["platform_is_external"] = swag.BoolValue(params.ClusterUpdateParams.Platform.IsExternal)
+		if *params.ClusterUpdateParams.Platform.Type == models.PlatformTypeExternal {
+			if params.ClusterUpdateParams.Platform.External != nil && params.ClusterUpdateParams.Platform.External.PlatformName != nil {
+				updates["platform_external_platform_name"] = params.ClusterUpdateParams.Platform.External.PlatformName
+			}
+			if params.ClusterUpdateParams.Platform.External != nil && params.ClusterUpdateParams.Platform.External.CloudControllerManager != nil {
+				updates["platform_external_cloud_controller_manager"] = params.ClusterUpdateParams.Platform.External.CloudControllerManager
+			}
+		} else {
+			updates["platform_external_platform_name"] = nil
+			updates["platform_external_cloud_controller_manager"] = nil
+		}
```

Comment:
else condition is never necessary

Reply:
why? I need to clean-up the external settings when the platform type isn't `external`

Reply:
AFAIU something like this would be equivalent, but easier to follow IMO

```suggestion
func (b *bareMetalInventory) updatePlatformParams(params installer.V2UpdateClusterParams, updates map[string]interface{}, usages map[string]models.Usage) error {
	if params.ClusterUpdateParams.Platform == nil || common.PlatformTypeValue(params.ClusterUpdateParams.Platform.Type) == "" {
		return nil
	}

	updates = getUpdatesForPlatformParams(params, updates)
	err := b.providerRegistry.SetPlatformUsages(
		common.PlatformTypeValue(params.ClusterUpdateParams.Platform.Type), usages, b.usageApi)
	if err != nil {
		return fmt.Errorf("failed setting platform usages, error is: %w", err)
	}
	return nil
}

func getUpdatesForPlatformParams(params installer.V2UpdateClusterParams, updates map[string]interface{}) map[string]interface{} {
	updates["platform_type"] = params.ClusterUpdateParams.Platform.Type
	updates["platform_is_external"] = swag.BoolValue(params.ClusterUpdateParams.Platform.IsExternal)
        if *params.ClusterUpdateParams.Platform.Type != models.PlatformTypeExternal {
		updates["platform_external_platform_name"] = nil
		updates["platform_external_cloud_controller_manager"] = nil
		return updates
        }
	if params.ClusterUpdateParams.Platform.External != nil && params.ClusterUpdateParams.Platform.External.PlatformName != nil {
		updates["platform_external_platform_name"] = params.ClusterUpdateParams.Platform.External.PlatformName
	}
	if params.ClusterUpdateParams.Platform.External != nil && params.ClusterUpdateParams.Platform.External.CloudControllerManager != nil {
		updates["platform_external_cloud_controller_manager"] = params.ClusterUpdateParams.Platform.External.CloudControllerManager
	}
	return updates
}
```